### ISSUE TITLE
XGBoost: A Scalable Tree Boosting System

### ISSUE THREAD

**Issue by yujiariyasu** [yujiariyasu/paper_summary#2]
# URL
https://arxiv.org/pdf/1603.02754.pdf

# Overview
・XGBoostはGradient BoostingとRandom Forestsを組み合わせたアンサンブル学習
・高度にスケーラブルなend-to-endなtree-boosting-systemである
・理論的に正当化された重み付き分位点スケッチの提案
・コア外ツリー学習のための効果的なキャッシュ認識ブロック構造を提案

# Point
・損失関数に正則化項を足した
・勾配に加えてへッシアン(2階微分)を考慮して計算速度の向上
・決定木の分割時における分割候補点の探索アルゴリズムの改善(ここが一番の改善ポイント？lossから直接分岐スコアを求める手法を提案した)
・疎なデータに対する効率的なルーティング
・out-of-core(CPU外)での学習が可能。計算速度も高速化
・汎化性能を上げるためにlearning-rate(eta)を下げていき、その都度最適化を行う

# Algorithm
![スクリーンショット 2020-03-17 10 42 19](https://user-images.githubusercontent.com/28746788/76813675-0f2aa180-683c-11ea-9151-17a351072f45.png)
↓
![スクリーンショット 2020-03-17 10 43 15](https://user-images.githubusercontent.com/28746788/76813698-210c4480-683c-11ea-9f03-dfcc6330f032.png)
木の構造q(x)が固定のとき，Lの最適値は以下の`スコア関数`となる
![スクリーンショット 2020-03-17 10 43 43](https://user-images.githubusercontent.com/28746788/76813745-44cf8a80-683c-11ea-82ae-f4f17ddc741d.png)

以下はスパース性を考慮した分割の検出
![スクリーンショット 2020-03-17 11 05 24](https://user-images.githubusercontent.com/28746788/76814868-5ebe9c80-683f-11ea-832d-2a449d9c6529.png)

# Experiment
![スクリーンショット 2020-03-17 11 11 08](https://user-images.githubusercontent.com/28746788/76815106-089e2900-6840-11ea-9f51-1cec3dcf4d6f.png)
4つ目のデータは17億サンプル、1テラ以上。このデータはout-of-coreの実験でも使用された。(その際はAWSで実験)
<img width="358" alt="スクリーンショット 2020-03-17 17 37 18" src="https://user-images.githubusercontent.com/28746788/76837484-fe971d00-6875-11ea-9764-2efc0b65c228.png">

使用したマシン：a Dell PowerEdge R420 with two eight-core
Intel Xeon (E5-2470) (2.3GHz) and 64GB of memory

# Next
LightGBM: A Highly Efficient Gradient Boosting Decision Tree
